hera_pspec/pspecdata.py

```diff
@@ -1225,6 +1279,204 @@ def get_unnormed_V(self, key1, key2, model='empirical'):
 
         return auto_term + cross_term
 
+    def get_analytic_covariance(self, key1, key2, M, model='time_average', cov_types=['original']):
+        """
+        Calculates the auto-covariance matrix for both the real and imaginary
+        parts of bandpowers (i.e., the q vectors and the p vectors). 
+
+        Define
+        
+        .. math ::
+            Real part of q_a = (1/2) (q_a + q_a^*)
+            Imaginary part of q_a = (1/2i) (q_a - q_a^dagger) 
+            Real part of p_a = (1/2) (p_a + p_a^dagger)
+            Imaginary part of p_a = (1/2i) (p_a - p_a^dagger)
+
+            E^{12,a} = (1/2) R_1 Q^a R_2
+            C^{12} = <x1 x2^dagger> - <x1><x2^dagger>
+            P^{12} = <x1 x2> - <x1><x2>
+            S^{12} = <x1^* x2^*> - <x1^*> <x2^*>
+            p_a = M_{ab} q_b
+
+        Then
+        
+        .. math ::
+        The variance of (1/2) (q_a + q_a^dagger):
+        (1/4){ (<q_a q_a> - <q_a><q_a>) + 2(<q_a q_a^dagger> - <q_a><q_a^dagger>) 
+        + (<q_a^dagger q_a^dagger> - <q_a^dagger><q_a^dagger>) }
+
+        The variance of (1/2i) (q_a - q_a^dagger) :
+        (-1/4){ (<q_a q_a> - <q_a><q_a>) - 2(<q_a q_a^dagger> - <q_a><q_a^dagger>) 
+        + (<q_a^dagger q_a^dagger> - <q_a^dagger><q_a^dagger>) }
+
+        The variance of (1/2) (p_a + p_a^dagger):
+        (1/4) { M_{ab} M_{ac} (<q_b q_c> - <q_b><q_c>) + 
+        M_{ab} M_{ac}^* (<q_b q_c^dagger> - <q_b><q_c^dagger>) + 
+        M_{ab}^* M_{ac} (<q_b^dagger q_c> - <q_b^dagger><q_c>) + 
+        M_{ab}^* M_{ac}^* (<q_b^dagger q_c^dagger> - <q_b^dagger><q_c^dagger>) }
+
+        The variance of (1/2i) (p_a - p_a^dagger):
+        (-1/4) { M_{ab} M_{ac} (<q_b q_c> - <q_b><q_c>) - 
+        M_{ab} M_{ac}^* (<q_b q_c^dagger> - <q_b><q_c^dagger>) - 
+        M_{ab}^* M_{ac} (<q_b^dagger q_c> - <q_b^dagger><q_c>) + 
+        M_{ab}^* M_{ac}^* (<q_b^dagger q_c^dagger> - <q_b^dagger><q_c^dagger>) }
+
+        where
+        <q_a q_b> - <q_a><q_b> = 
+                    tr(E^{12,a} C^{21} E^{12,b} C^{21})
+                    + tr(E^{12,a} P^{22} E^{21,b*} S^{11})
+        <q_a q_b^dagger> - <q_a><q_b^dagger> =          
+                    tr(E^{12,a} C^{22} E^{21,b} C^{11})
+                    + tr(E^{12,a} P^{21} E^{12,b *} S^{21})
+        <q_a^dagger q_b^dagger> - <q_a^dagger><q_b^dagger> =            
+                    tr(E^{21,a} C^{12} E^{21,b} C^{12})
+                    + tr(E^{21,a} P^{11} E^{12,b *} S^{22})
+
+        Note that
+        
+        .. math ::
+            E^{12,a}_{ij}.conj = E^{21,a}_{ji}
+
+        This function estimates C^1, C^2, P^{12}, and S^{12} empirically by 
+        default. (So while the pointy brackets <...> should in principle be 
+        ensemble averages, in practice the code performs averages in time.)
+
```

Comment:
I think the doc-string comment also should be inserted here, above the `Parameters` line.